ReactAndroid/src/main/java/com/facebook/react/uimanager/UIImplementation.java

```diff
@@ -357,6 +368,142 @@ protected void handleUpdateView(
+        int moveFromIndex = moveFrom.getInt(i);
+        int tagToMove = cssNodeToManage.getChildAt(moveFromIndex).getReactTag();
+        viewsToAdd[i] = new ViewAtIndex(
+          tagToMove,
+          moveTo.getInt(i));
+        indicesToRemove[i] = moveFromIndex;
+        tagsToRemove[i] = tagToMove;
+      }
+    }
+
+    if (numToAdd > 0) {
+      Assertions.assertNotNull(addChildTags);
+      Assertions.assertNotNull(addAtIndices);
+      for (int i = 0; i < numToAdd; i++) {
+        int viewTagToAdd = addChildTags.getInt(i);
+        int indexToAddAt = addAtIndices.getInt(i);
+        viewsToAdd[numToMove + i] = new ViewAtIndex(viewTagToAdd, indexToAddAt);
+      }
+    }
+
+    if (numToRemove > 0) {
+      Assertions.assertNotNull(removeFrom);
+      for (int i = 0; i < numToRemove; i++) {
+        int indexToRemove = removeFrom.getInt(i);
+        int tagToRemove = cssNodeToManage.getChildAt(indexToRemove).getReactTag();
+        indicesToRemove[numToMove + i] = indexToRemove;
+        tagsToRemove[numToMove + i] = tagToRemove;
+        tagsToDelete[i] = tagToRemove;
+      }
+    }
+
+    // NB: moveFrom and removeFrom are both relative to the starting state of the View's children.
+    // moveTo and addAt are both relative to the final state of the View's children.
+    //
+    // 1) Sort the views to add and indices to remove by index
+    // 2) Iterate the indices being removed from high to low and remove them. Going high to low
+    //    makes sure we remove the correct index when there are multiple to remove.
+    // 3) Iterate the views being added by index low to high and add them. Like the view removal,
+    //    iteration direction is important to preserve the correct index.
+
+    Arrays.sort(viewsToAdd, ViewAtIndex.COMPARATOR);
+    Arrays.sort(indicesToRemove);
+
+    // Apply changes to CSSNodeDEPRECATED hierarchy
+    int lastIndexRemoved = -1;
+    for (int i = indicesToRemove.length - 1; i >= 0; i--) {
+      int indexToRemove = indicesToRemove[i];
+      if (indexToRemove == lastIndexRemoved) {
+        throw new IllegalViewOperationException("Repeated indices in Removal list for view tag: "
+          + viewTag);
+      }
+      cssNodeToManage.removeChildAt(indicesToRemove[i]);
+      lastIndexRemoved = indicesToRemove[i];
+    }
+    ArrayList<ViewAtIndex> vais = new ArrayList<>(Arrays.asList(viewsToAdd));
+    ArrayList<ViewAtIndex> illegals = new ArrayList<>();
+    for (int i = 0; i < vais.size(); i++) {
+      ViewAtIndex viewAtIndex = vais.get(i);
+      ReactShadowNode cssNodeToAdd = mShadowNodeRegistry.getNode(viewAtIndex.mTag);
+      if (cssNodeToAdd == null) {
+        throw new IllegalViewOperationException("Trying to add unknown view tag: "
+          + viewAtIndex.mTag);
+      }
+      // If there is a RCTRawText child node, and the parent is not a ReactBaseTextShadowNode, donot add the illegal
+      // RCTRawText childNode to parent node.
+      // Fix bug 1216685-IndexOutOfBoundsException
```

Comment:
You can remove the "fix bug" line here, as it should be implied by the commit message.

Can you fix the "donot" typo while you're at it?